tests/z_local_loopback_test.c

```diff
@@ -0,0 +1,245 @@
+//
```

Comment:
The tests covers basic scenarios:
  - Single local subscriber
  - Single local queryable

It would be nice to have also:
  - Multiple local subscribers/queriables
  - Mixed local + remote subscribers/queriables